### ISSUE TITLE
Unable to compile project that links to atls.lib

### ISSUE THREAD

**Issue by cmconti** [actions/virtual-environments#1020]
**Describe the bug**
We have a number of C++ projects that link to atls.lib, are built using the v141 toolset, and are not using the spectre mitigations.  Starting with 20200531.1, compilation now fails with the error:
`##[error]LINK(0,0): Error LNK1104: cannot open file 'atls.lib'`

This worked previously with 20200517.1
Further investigation shows that the installed VS2019 image appears to be missing the components:
- Microsoft.VisualStudio.Component.VC.v141.ATL
- Microsoft.VisualStudio.Component.VC.v141.MFC

**Area for Triage**: C/C++

**Question, Bug, or Feature?**: Bug

**Virtual environments affected**
- [ ] macOS 10.15
- [ ] Ubuntu 16.04 LTS
- [ ] Ubuntu 18.04 LTS
- [ ] Ubuntu 20.04 LTS
- [ ] Windows Server 2016 R2
- [x] Windows Server 2019

**Expected behavior**
Expect project to compile

**Actual behavior**
Attached a zip of a sample project (just an empty wizard-generated ATL exe) and a zip of the pipeline logs.

[Az_ATLS.zip](https://github.com/actions/virtual-environments/files/4748093/Az_ATLS.zip)
[logs_1922.zip](https://github.com/actions/virtual-environments/files/4748094/logs_1922.zip)

**Comment by miketimofeev** [comment]
Hi @cmconti! These workloads were supposed to be installed during the build
https://github.com/actions/virtual-environments/releases/tag/win19%2F20200531.1
Could you please try to install them manually by adding this as the first step of your workflow? Does it help?
```
jobs:
  add workflows:
    runs-on: windows-latest
    steps:
      - uses: actions/checkout@v2
      - name: add workload
        run: |
          cd "C:\Program Files (x86)\Microsoft Visual Studio\Installer\"
          $InstallPath = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise"
          $WorkLoads = '--add Microsoft.VisualStudio.Component.VC.v141.ATL.ARM.Spectre '
          '--add Microsoft.VisualStudio.Component.VC.v141.ATL.ARM64.Spectre ' + `
          '--add Microsoft.VisualStudio.Component.VC.v141.ATL.Spectre ' + `
          '--add Microsoft.VisualStudio.Component.VC.v141.MFC.ARM.Spectre ' + `
          '--add Microsoft.VisualStudio.Component.VC.v141.MFC.ARM64.Spectre ' + `
          '--add Microsoft.VisualStudio.Component.VC.v141.MFC.Spectre '
          $Arguments = ('/c', "vs_installer.exe", 'modify', '--installPath', "`"$InstallPath`"", $WorkLoads, '--quiet', '--norestart', '--nocache')
          $process = Start-Process -FilePath cmd.exe -ArgumentList $Arguments -Wait -PassThru -WindowStyle Hidden
        shell: powershell
```

**Comment by cmconti** [comment]
I have not done much with multiple jobs, and can't find any examples of adding to the workflow.  What is the proper syntax for merging the above yml snippet into my pipeline file?

```yml
trigger:
- master

pool:
  vmImage: 'windows-latest'

variables:
  solution: '**/*.sln'
  buildPlatform: 'x86'
  buildConfiguration: 'Debug'

steps:
#- script : |
#    "C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installer.exe" export --installpath "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise" --config "%TEMP%\atls.vsconfig" --noweb --quiet
#    type "%TEMP%\atls.vsconfig"

- task: VSBuild@1
  inputs:
    solution: '$(solution)'
    platform: '$(buildPlatform)'
    configuration: '$(buildConfiguration)'
```

**Comment by miketimofeev** [comment]
@cmconti Oh, it's AzDO, right? Then you can use the following:
```
  - powershell: |
      cd "C:\Program Files (x86)\Microsoft Visual Studio\Installer\"
      $InstallPath = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise"
      $WorkLoads = '--add Microsoft.VisualStudio.Component.VC.v141.ATL.ARM.Spectre ' + `
      '--add Microsoft.VisualStudio.Component.VC.v141.ATL.ARM64.Spectre ' + `
      '--add Microsoft.VisualStudio.Component.VC.v141.ATL.Spectre ' + `
      '--add Microsoft.VisualStudio.Component.VC.v141.MFC.ARM.Spectre ' + `
      '--add Microsoft.VisualStudio.Component.VC.v141.MFC.ARM64.Spectre ' + `
      '--add Microsoft.VisualStudio.Component.VC.v141.MFC.Spectre'
      $Arguments = ('/c', "vs_installer.exe", 'modify', '--installPath', "`"$InstallPath`"", $WorkLoads, '--quiet', '--norestart', '--nocache')
      $process = Start-Process -FilePath cmd.exe -ArgumentList $Arguments -Wait -PassThru -WindowStyle Hidden
      $process.ExitCode
```
It will take about 10-15 min. Exit code should be 0

**Comment by cmconti** [comment]
vs_installer exits with 1 after less than a minute.  Tried adding --noupdateinstaller to the params.  same result.

**Comment by miketimofeev** [comment]
@cmconti it may be some issue with copy-paste. Could you try this yaml in zip below? It works for me

[Update.zip](https://github.com/actions/virtual-environments/files/4754321/Update.zip)

**Comment by cmconti** [comment]
Exit code is still 1.  See attached log:
[logs_1945.zip](https://github.com/actions/virtual-environments/files/4754750/logs_1945.zip)

**Comment by cmconti** [comment]
Side note:  earlier, I had dumped the .vsconfig from a 20200531.1 image via the script lines:
"C:\Program Files (x86)\Microsoft Visual Studio\Installer\vs_installer.exe" export --installpath "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise" --config "%TEMP%\atls.vsconfig" --noweb --quiet
type "%TEMP%\atls.vsconfig"

would it be possible to do the same from a 20200517.1 image?  I don't think I can run the older image anymore, but it would be interesting to compare the results.  I'm wondering if the issue is a change in the 'recommended' components between VS 2019 16.5 and 16.6...  If this is something I can try, what is the syntax for specifying the older image?

**Comment by miketimofeev** [comment]
@cmconti pretty strange that installation didn't work in the runtime, I tried it in AzDO also, but with a newer image with VS 16.6.3 that is being rolled out at the moment(https://github.com/actions/virtual-environments/releases/tag/win19%2F20200608.1)
Unfortunately, we can't do the same with a 20200517.1 right now because it'll take hours to deploy the image to the test environment, but I've compared results between 0531.1 and 0608.1 and they are the same. v141.ATL and v141.MFC both presented. 
We're also thinking about Visual Studio breaking changes and trying to reproduce the issue from your sample at the moment.
Thank you for your patience!
